### PR TITLE
MINOR: Improve API docs of (flatT|t)ransform

### PR DESCRIPTION
**Pull request by cadonna**
This commit is a follow-up of pull request #5273.

- Adds and reformulates some parts
- Removes parameters from links to methods in the javadocs 

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: streams/src/main/java/org/apache/kafka/streams/kstream/KStream.java [hunk 1]
```diff
@@ -509,11 +509,11 @@ void to(final TopicNameExtractor<K, V> topicExtractor,
      *
      * KStream outputStream = inputStream.transform(new TransformerSupplier() { ... }, "myTransformState");
      * }</pre>
-     * Within the {@link Transformer}, the state is obtained via the {@link  ProcessorContext}.
+     * Within the {@link Transformer}, the state is obtained via the {@link ProcessorContext}.
      * To trigger periodic actions via {@link org.apache.kafka.streams.processor.Punctuator#punctuate(long) punctuate()},
      * a schedule must be registered.
      * The {@link Transformer} must return a {@link KeyValue} type in {@link Transformer#transform(Object, Object)
-     * transform()} and {@link org.apache.kafka.streams.processor.Punctuator#punctuate(long) punctuate()}.
+     * transform(K, V)} and {@link org.apache.kafka.streams.processor.Punctuator#punctuate(long) punctuate(long)}.
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/kstream/KStream.java]: We don't add parameters in any JavaDocs. We use `()` in purpose to "shorten" long parameter lists (if usr follows the link, she sees the full parameter list). Note that
```
{@link org.apache.kafka.streams.processor.Punctuator#punctuate(long)}
```
and
```
{@link org.apache.kafka.streams.processor.Punctuator#punctuate(long) punctuate(long)}
```
would be the same (both render as `punctuate(long)`), but we want to get simplified `punctuate()`.

Similar below.

**Review by mjsax — COMMENTED** [review]
Thanks for the follow up @cadonna! One clarification to my original comment.

File: streams/src/main/java/org/apache/kafka/streams/kstream/KStream.java (hunk 1 above)
Comment (cadonna) [reply streams/src/main/java/org/apache/kafka/streams/kstream/KStream.java]: I removed the parameters from the links in the parts that I modified. However, in the other parts there are also links with parameters, e.g., line 544. Should I also change those in this PR?

Comment (mjsax) [reply streams/src/main/java/org/apache/kafka/streams/kstream/KStream.java]: I was just looking across the JavaDocs. Seems we are not very consistent in what we do...  L544 is not the only example. I leave it up to you if you want to clean it up, or not. We also just merge this PR as-is. Might not be worth the time?

Comment (cadonna) [reply streams/src/main/java/org/apache/kafka/streams/kstream/KStream.java]: I removed the parameters from all the links to methods in the javadocs for `transform` and `flatTransform`.

**Review by mjsax — APPROVED** [review]
LGTM.

Call for second review (any of @guozhangwang @bbejeck @vvcephei @ableegoldman)

**Review by bbejeck — APPROVED** [review]
LGMT

**Comment by bbejeck** [comment]
Merged #6365 into trunk.

Thanks for the followup PR @cadonna!

**Pull request merged**